runtime.md

```diff
@@ -1,5 +1,47 @@
 # Runtime and Lifecycle
 
+## State
+
+The runtime state for a container is persisted on disk so that external tools can consume and act on this information.
+The runtime state is stored in a JSON encoded file.
+It is recommended that this file is stored in a temporary filesystem so that it can be removed on a system reboot.
+On Linux based systems the state information should be stored in `/run/oci/containers`.
+The directory structure for a container is `/run/oci/containers/<containerID>/state.json`.
+By providing a default location that container state is stored external applications can find all containers running on a system.
+
+* **version** (string) Version of the OCI specification used when creating the container.
+* **id** (string) ID is the container's ID.
+* **pid** (int) Pid is the ID of the main process within the container.
+* **root** (string) Root is the path to the container's bundle directory.
+
+The ID is provided in the state because hooks will be executed with the state as the payload.
+This allows the hook to perform clean and teardown logic after the runtime destroys it's own state.
+The root directory to the bundle is provided in the state so that consumers can find the container's configuration and rootfs.
+
+*Example*
+
+```json
+{
+    "id": "oci-container",
+    "pid": 4422,
+    "root": "/containers/redis"
+}
+```
+
+Linux systems add some platform specific information to the state.
+
+* **externalFds** Paths to the container's open file descriptors.
+
+*Example Linux*
+
+```json
+{
+    "externalFds": [
+        "/proc/33/fd/1"
```

Comment:
For example, with one process piping its stdout into another's stdin, I see:

    $ ls -l /proc/19119/fd/1 /proc/19120/fd/0
    l-wx------ 1 wking wking 64 Aug 26 15:38 /proc/19119/fd/1 -> pipe:[83842]
    lr-x------ 1 wking wking 64 Aug 26 15:38 /proc/19120/fd/0 -> pipe:[83842]

You can also find the pipe name *and* the associated process at the other end using `lsof`:

    $ lsof -p 19120 +E | grep 0r
    cat     19119 wking    1w  FIFO   0,10      0t0    83842 pipe 19120,less,0r
    less    19120 wking    0r  FIFO   0,10      0t0    83842 pipe 19119,cat,1w

Since the CRIU bindings can pull this information from the kernel, and I'd suggest not supplying it in the state JSON.